gateway-service/src/main/java/org/zowe/apiml/gateway/config/DiscoveryClientConfig.java

```diff
@@ -14,46 +14,87 @@
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 /**
  * This configuration override bean EurekaClient with custom ApimlDiscoveryClient. This bean offer additional method
  * fetchRegistry. User can call this method to asynchronously fetch new data from discovery service. There is no time
  * to fetching.
  * <p>
- * Configuration also add listeners to call other beans waiting for fetch new registry. It speed up distribution of
+ * Configuration also add listeners to call other beans waiting for fetch new registry. It speeds up distribution of
  * changes in whole gateway.
  */
 @Configuration
 @RequiredArgsConstructor
 public class DiscoveryClientConfig {
     private final ApplicationContext context;
     private final AbstractDiscoveryClientOptionalArgs<?> optionalArgs;
+    private final EurekaJerseyClientImpl.EurekaJerseyClientBuilder eurekaJerseyClientBuilder;
+
+    @Value("${apiml.service.centralRegistryUrls:-}")
+    private String[] centralRegistryUrls;
 
     @Bean(destroyMethod = "shutdown")
     @RefreshScope
-    public ApimlDiscoveryClient eurekaClient(ApplicationInfoManager manager,
-                                             EurekaClientConfig config,
-                                             @Autowired(required = false) HealthCheckHandler healthCheckHandler
+    public ApimlDiscoveryClient primaryApimlEurekaClient(ApplicationInfoManager manager,
+                                                         EurekaClientConfig config,
+                                                         @Autowired(required = false) HealthCheckHandler healthCheckHandler
     ) {
-        ApplicationInfoManager appManager;
-        if (AopUtils.isAopProxy(manager)) {
-            appManager = ProxyUtils.getTargetObject(manager);
-        } else {
-            appManager = manager;
-        }
+        ApplicationInfoManager appManager = AopUtils.isAopProxy(manager) ? ProxyUtils.getTargetObject(manager) : manager;
+
         final ApimlDiscoveryClient discoveryClientClient = new ApimlDiscoveryClient(appManager, config, this.optionalArgs, this.context);
         discoveryClientClient.registerHealthCheck(healthCheckHandler);
 
         return discoveryClientClient;
     }
 
+    @Bean(destroyMethod = "shutdown")
+    @ConditionalOnProperty(name = "apiml.service.centralRegistryUrls")
+    @RefreshScope
+    public DiscoveryClientWrapper additionalDiscoverClientWrapper(ApplicationInfoManager manager,
+                                                                  EurekaClientConfig config,
+                                                                  @Autowired(required = false) HealthCheckHandler healthCheckHandler
+    ) {
+        ApplicationInfoManager appManager = AopUtils.isAopProxy(manager) ? ProxyUtils.getTargetObject(manager) : manager;
+        List<ApimlDiscoveryClient> listOfDiscoveryClients = new ArrayList<>();
+
+        for (String url : centralRegistryUrls) {
```

Comment:
Shall we implement also different routing for each DS? See requirement for multitenancy https://github.com/zowe/api-layer/blob/142716baea6363dc81c5a3808a7035dc05ca008e/cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/service/RouteLocator.java#L105

Reply:
this feature is only to replace static registration of domain-level GWs, routing should be addressed in separate PR